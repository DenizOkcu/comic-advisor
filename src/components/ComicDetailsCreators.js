import React from "react";

const ComicDetailsCreators = props => {
  const creators = props.creators.map(creator => (
    <li key={creator.name}>{`${creator.name}(${creator.role})`}</li>
  ));

  return (
    <section>
      Created by:
      <ul>{creators}</ul>
    </section>
  );
};

export default ComicDetailsCreators;
