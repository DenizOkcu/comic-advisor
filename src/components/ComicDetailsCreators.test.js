import React from "react";
import { shallow } from "enzyme";
import ComicDetailsCreators from "./ComicDetailsCreators";

describe("ComicDetailsCreators", () => {
  it('should render correctly in "debug" mode', () => {
    const creators = [{ id: 0, name: "writer name", role: "writer" }];
    const component = shallow(
      <ComicDetailsCreators creators={creators} debug />
    );

    expect(component).toMatchSnapshot();
  });
});
