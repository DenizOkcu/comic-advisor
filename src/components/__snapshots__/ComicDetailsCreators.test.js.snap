// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicDetailsCreators should render correctly in "debug" mode 1`] = `
<section>
  Created by:
  <ul>
    <li
      key="writer name"
    >
      writer name(writer)
    </li>
  </ul>
</section>
`;
